### PR TITLE
Rework management of uploaded files

### PR DESCRIPTION
**Pull request by Thomah**
# What this PR does / why we need it:

It improves storage of images by using workspaces for projects, ideas and news. The client defines a workspace for a project, ideas or news (with a UUID for example) and all uploaded images will be created in a folder with the workspace's name.

It provides an endpoint for consulting images. It will allow us to unmount folder of images from the web component and gain stability with the Kubernetes deployment.

Endpoint for upload and deletion of images are compatible with Summernote editor.

The content can now only be owned by a single organization. It simplify the code a lot and more entities will change that way. It we need to share entities between organizations, we will add a sharing feature.

There also is a small cleanup of dead code such as Tags which will be re-implemented later.

Finally I also included a PR template for future contributions.

Fixes :
- https://github.com/les-projets-cagnottes/backlog/issues/38
- https://github.com/les-projets-cagnottes/backlog/issues/32

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Code compilation
- [x] All tests passing
- [x] Extended the documentation, if applicable

### CODE REVIEW

File: src/main/java/fr/lesprojetscagnottes/core/common/strings/StringGenerator.java [hunk 1]
```diff
@@ -14,4 +22,10 @@ public static String randomString() {
         return builder.toString();
     }
 
+    public static String imageName() {
+        SimpleDateFormat df = new SimpleDateFormat("yyyyMMdd_HHmmss");
```
Comment (jeremylegendre44) [inline src/main/java/fr/lesprojetscagnottes/core/common/strings/StringGenerator.java:26]: Pourquoi ne pas utiliser la dernière API pour les dates plutôt que Clock ? 
 LocalDateTime date = LocalDateTime.now();
 DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd_HHmmss");

File: src/main/java/fr/lesprojetscagnottes/core/common/security/TokenProvider.java [hunk 2]
```diff
@@ -89,12 +86,12 @@ public String generateToken(Authentication authentication) {
...
 
     public String refreshToken(String token) {
-        final Date createdDate = clock.now();
+        final Date createdDate = new Date();
         final Date expirationDate = new Date(createdDate.getTime() + ACCESS_TOKEN_VALIDITY_SECONDS * 1000);
 
         SimpleDateFormat dt1 = new SimpleDateFormat("yyyyy-MM-dd");
-        LOGGER.debug("createdDate = " + dt1.format(createdDate));
-        LOGGER.debug("expirationDate = " + dt1.format(expirationDate));
+        log.debug("createdDate = " + dt1.format(createdDate));
```
Comment (jeremylegendre44) [inline src/main/java/fr/lesprojetscagnottes/core/common/security/TokenProvider.java:93]: il manque la même correction que dans le fichier StringGenerator.java pour la date

Comment (Thomah) [reply src/main/java/fr/lesprojetscagnottes/core/common/security/TokenProvider.java:93]: Ici, pour garder les choses simples, on est obligé de garder l'ancienne API car la library JWT manipule des objets java.util.Date

Comment (jeremylegendre44) [reply src/main/java/fr/lesprojetscagnottes/core/common/security/TokenProvider.java:93]: Tu peux retransformer en Date ensuite si jamais : 
// LocalDateTime -> Date
LocalDateTime localDateTime = LocalDateTime.of(2020,2,20,21,46,31);
Date date2 = Date.from(localDateTime.atZone(ZoneId.systemDefault()).toInstant());

Comment (Thomah) [reply src/main/java/fr/lesprojetscagnottes/core/common/security/TokenProvider.java:93]: Je prends note mais vu le cas d'usage je ne vois pas l'intérêt de passer par une conversion qui coute du CPU. Sans compter le fait que la méthode `refreshToken` est une des méthodes les plus appelées via l'API alors dans l'idéal il faut être assez optimisé 😄 
Autant fournir directement les objets Date attendus par la library JWT non ?

Comment (jeremylegendre44) [reply src/main/java/fr/lesprojetscagnottes/core/common/security/TokenProvider.java:93]: Après niveau CPU la conversion ne doit pas coûter énormément je pense mais dans le doute...

**Pull request merged**